src/index.js

```diff
@@ -1,198 +1,49 @@
 'use strict'
 
-const debug = require('debug')
-const log = debug('libp2p:tcp')
-const tcp = require('net')
-const multiaddr = require('multiaddr')
-const Address6 = require('ip-address').Address6
+const net = require('net')
+const toPull = require('stream-to-pull-stream')
 const mafmt = require('mafmt')
-// const parallel = require('run-parallel')
 const contains = require('lodash.contains')
-const os = require('os')
+const isFunction = require('lodash.isfunction')
+const noop = require('lodash.noop')
 const Connection = require('interface-connection').Connection
+const debug = require('debug')
+const log = debug('libp2p:tcp')
 
-exports = module.exports = TCP
-
-const IPFS_CODE = 421
-const CLOSE_TIMEOUT = 2000
-
-function TCP () {
-  if (!(this instanceof TCP)) {
-    return new TCP()
-  }
+const createListener = require('./listener')
 
-  this.dial = function (ma, options, callback) {
-    if (typeof options === 'function') {
+module.exports = class TCP {
+  dial (ma, options, callback) {
+    if (isFunction(options)) {
       callback = options
       options = {}
     }
 
     if (!callback) {
-      callback = function noop () {}
+      callback = noop
```

Comment:
It would be actually a good stat to have, how many lines of code should a module be to justify not writing it by yourself